include/eve/module/core/regular/impl/simd/ppc/any.hpp

```diff
@@ -10,27 +10,44 @@
 #include <eve/concept/value.hpp>
 #include <eve/detail/implementation.hpp>
 #include <eve/module/core/constant/true.hpp>
+#include <eve/module/core/regular/simd_cast.hpp>
 
 namespace eve::detail
 {
 template<callable_options O, arithmetic_scalar_value T, typename N>
 EVE_FORCEINLINE bool
 any_(EVE_REQUIRES(vmx_), O const& opts, logical<wide<T, N>> const& v0) noexcept requires ppc_abi<abi_t<T, N>>
 {
-  if constexpr (!match_option<condition_key, O, ignore_none_>) return any.behavior(cpu_{}, opts, v0);
-  auto m = v0.bits();
+  if constexpr( N::value == 1 ) {
+    auto m = v0.bits();
 
-  if constexpr( N::value == 1 ) { return static_cast<bool>(m.get(0)); }
-  else if constexpr( N::value == expected_cardinal_v<T, ppc_> )
-  {
-    return vec_any_eq(m.storage(), true_(eve::as(v0)).storage());
+    if constexpr (match_option<condition_key, O, ignore_none_>)
+    {
+      return static_cast<bool>(m.get(0));
+    }
+    else
+    {
+      const auto cm = expand_mask(opts[condition_key], as<wide<T, N>>{});
+      return static_cast<bool>(m.get(0)) && static_cast<bool>(cm.get(0));
+    }
   }
   else
   {
-    logical<wide<T>> mm = [](auto i, auto) { return i < N::value; };
-    m &= bit_cast(mm, as<logical<wide<T, N>>>()).bits();
+    auto m = simd_cast(v0, as<logical<wide<T>>>{}).bits();
+
+    if constexpr (N::value != expected_cardinal_v<T, ppc_>)
+    {
+      logical<wide<T>> mm = [](auto i, auto) { return i < N::value; };
+      m &= mm.bits();
+    }
+
+    if constexpr (!match_option<condition_key, O, ignore_none_>)
+    {
+      const auto cm = simd_cast(expand_mask(opts[condition_key], as<wide<T, N>>{}), as<logical<wide<T>>>{});
+      m &= cm.bits();
+    }
```

Comment:
Hmm. Hmm. Hmm.

Here is what I don't like.

* If we have ignore none - we should remove garbage from the main register.
* If we have relative conditional - expand_mask from that could for free generate False everywhere where garbage might go
* If we have a non-relative conditional - expand_mask won't be able to zero out false elements for free.

^ We need some helper to handle all of this stuff.